src/compiler/diagnosticMessages.json

```diff
@@ -1647,6 +1647,14 @@
         "category": "Error",
         "code": 2517
     },
+    "A 'this'-based type guard is not assignable to a parameter-based type guard": {
+        "category": "Error",
+        "code": 2518
+    },
+    "A 'this'-based type predicate is only allowed in class or interface members, get accessors, or return type positions for functions and methods": {

```

Comment:
"A 'this'-based type predicate is only allowed as the return type of functions, methods, and get accessors, or on members of classes and interfaces."